src/drafter.cc

```diff
@@ -0,0 +1,244 @@
+// vi:cin:et:sw=4 ts=4
+//
+//  drafter.cc
+//
+//  Created by Jiri Kratochvil on 2015-02-11
+//  Copyright (c) 2015 Apiary Inc. All rights reserved.
+//
+//
+//
+
+#include <string>
+#include <iostream>
+#include <sstream>
+#include <fstream>
+
+#include <tr1/memory>
+#include <tr1/shared_ptr.h>
+
+#include "cmdline.h"
+
+#include "snowcrash.h"
+#include "SectionParserData.h"  // snowcrash::BlueprintParserOptions
+
+#include "sos.h"
+#include "sosJSON.h"
+#include "sosYAML.h"
+
+#include "SerializeAST.h"
+#include "SerializeSourcemap.h"
+
+namespace config {
+    static const std::string Program        = "drafter";
+
+    static const std::string Output         = "output";
+    static const std::string Format         = "format";
+    static const std::string Render         = "render";
+    static const std::string Sourcemap      = "sourcemap";
+    static const std::string Validate       = "validate";
+    static const std::string Version        = "version";
+    static const std::string UseLineNumbers = "use-line-num";
+};
+
+struct Config {
+    std::string input;
+    bool lineNumbers;
+    bool validate;
+    std::string format;
+    std::string sourceMap;
+    std::string output;
+};
+
+void PrepareCommanLineParser(cmdline::parser& parser)
+{
+    parser.set_program_name(config::Program);
+
+    parser.add<std::string>(config::Output,    'o', "save output AST into file", false);
+    parser.add<std::string>(config::Format,    'f', "output AST format", false, "yaml", cmdline::oneof<std::string>("yaml", "json"));
+    parser.add<std::string>(config::Sourcemap, 's', "export sourcemap AST into file", false);
+    parser.add("help",                         'h', "display this help message");
+    parser.add(config::Version ,               'v', "print Drafter version");
+    parser.add(config::Validate,               'l', "validate input only, do not print AST");
+    parser.add(config::UseLineNumbers ,        'u', "use line and row number instead of character index when printing annotation");
+
+    std::stringstream ss;
+
+    ss << "<input file>\n\n";
+    ss << "API Blueprint Parser\n";
+    ss << "If called without <input file>, 'drafter' will listen on stdin.\n";
+
+    parser.footer(ss.str());
+}
+
+#define DRAFTER_VERSION_STRING "x.x.x"
```

Comment:
Shouldn't this be at the beginning of the file after `#includes`? Just curious if there is any reason for it be in the middle here.